core/src/test/java/org/apache/iceberg/hadoop/HadoopFileIOTest.java

```diff
@@ -111,16 +110,16 @@ public void testDeletePrefix() {
               hadoopFileIO.deletePrefix(scalePath.toUri().toString());
 
               // Hadoop filesystem will throw if the path does not exist
-              assertThrows(
-                  UncheckedIOException.class,
-                  () -> hadoopFileIO.listPrefix(scalePath.toUri().toString()).iterator());
+              Assertions.assertThatThrownBy(
+                      () -> hadoopFileIO.listPrefix(scalePath.toUri().toString()).iterator())
+                  .isInstanceOf(UncheckedIOException.class);
```

Comment:
can we add a `.hasMessage(..)` check here and below?

Reply:
I'll add the `.hasmessage` .

Reply:
Based on your review, I added the hasMessage in `Assertions.assertThatThrownBy` .